### ISSUE TITLE
Encryption : Call on Null while encrypt-all

### ISSUE THREAD

**Issue by ArthurGarnier** [owncloud/core#22386]
### Steps to reproduce
1. Enable default encryption module
2. Set up recovery key
3. On your server : sudo -u www-data php occ encryption:encrypt-all

### Expected behaviour

All the file should be encrypted

### Actual behaviour

Error message appear after several file : Call to a member function get() on null in /var/www/owncloud/apps/files_sharing/lib/updater.php on line 76

Observation : Run several times the command permits to encrypt more files.

### Server configuration
**Operating system**: Debian Jessie (Testing)

**Web server:** Apache 2.4.18-1

**Database:** Postgres 9.5

**PHP version:** php5 5.6.17

**ownCloud version:** 8.2.2

**Updated from an older ownCloud or fresh install:** Updated Older

**Signing status (ownCloud 9.0 and above):**

```
Login as admin user into your ownCloud and access 
http://example.com/index.php/settings/integrity/failed  ==> I'm redirected to /apps/files
paste the results here.
```

**List of activated apps:** 

```
Enabled:
  - activity: 2.1.3
  - calendar: 0.8.1
  - documents: 0.11.1
  - encryption: 1.1.0
  - files: 1.2.0
  - files_pdfviewer: 0.7
  - files_sharing: 0.7.0
  - files_texteditor: 2.0
  - files_versions: 1.1.0
  - files_videoviewer: 0.1.3
  - firstrunwizard: 1.1
  - gallery: 14.2.0
  - mail: 0.3.0
  - notifications: 0.1.0
  - provisioning_api: 0.3.0
  - templateeditor: 0.1

```

**The content of config/config.php:**

```
{
    "system": {
        "instanceid": "oca2dbc106e8",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "arthurgarnier.fr"
        ],
        "datadirectory": "\/home\/owncloud\/data",
        "overwrite.cli.url": "https:\/\/arthurgarnier.fr\/owncloud",
        "dbtype": "pgsql",
        "version": "8.2.2.2",
        "dbname": "owncloud",
        "dbhost": "localhost",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "logtimezone": "UTC",
        "installed": true,
        "theme": "",
        "maintenance": false,
        "loglevel": 2,
        "trashbin_retention_obligation": "auto",
        "mail_smtpmode": "sendmail",
        "mail_from_address": "owncloud",
        "mail_smtphost": "localhost",
        "mail_smtpport": "25",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "mail_smtpsecure": "tls",
        "singleuser": true
    }
}
```

**Are you using external storage, if yes which one:** local/smb/sftp/... No

**Are you using encryption:** No (hope yes... ;))

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/... No

### Client configuration
**Browser:** Chrome (latest)

**Operating system:** Arch Linux

#### Web server error log
```
No error in the log when the error in owncloud occur
```

#### ownCloud log (data/owncloud.log)
```
{"reqId":"*****************","remoteAddr":"","app":"no app in context","message":"Can not decrypt this file, probably this is a shared file. Please ask the file owner to reshare the file with you.","level":3,"time":"2016-02-15T11:54:39+00:00"}
{"reqId":"******************","remoteAddr":"","app":"no app in context","message":"Can not decrypt this file, probably this is a shared file. Please ask the file owner to reshare the file with you.","level":3,"time":"2016-02-15T11:54:41+00:00"}
{"reqId":"*****************","remoteAddr":"","app":"no app in context","message":"Can not decrypt this file, probably this is a shared file. Please ask the file owner to reshare the file with you.","level":3,"time":"2016-02-15T11:54:44+00:00"}
{"reqId":"*****************","remoteAddr":"","app":"no app in context","message":"Can not decrypt this file, probably this is a shared file. Please ask the file owner to reshare the file with you.","level":3,"time":"2016-02-15T11:54:45+00:00"}
{"reqId":"******************","remoteAddr":"","app":"PHP","message":"Call to a member function get() on null at \/var\/www\/owncloud\/apps\/files_sharing\/lib\/updater.php#76","level":3,"time":"2016-02-15T11:54:47+00:00"}

```

**Comment by PVince81** [comment]
Please use the issue template: https://raw.githubusercontent.com/owncloud/core/master/issue_template.md

Missing information:
- which ownCloud version are you using ?
- stack trace from log, if any

Also it looks like it fails when encrypted shared files ?

CC @schiesbn

**Comment by Neo-Burner** [comment]
Edited, sorry ;)

**Comment by butonic** [comment]
oc8.2.3

```
$ sudo -u www-data php occ  encryption:encrypt-all


You are about to start to encrypt all files stored in your ownCloud.
It will depend on the encryption module you use which files get encrypted.
Depending on the number and size of your files this can take some time
Please make sure that no user access his files during this process!

Do you really want to continue? (y/n) y


Encrypt all files with the Default encryption module
====================================================


Create key-pair for every user
------------------------------

This module will encrypt all files in the users files folder initially.
Already existing versions and files in the trash bin will not be encrypted.


 Key-pair created for all users 
 [>---------------------------]

Start to encrypt users files
----------------------------


 encrypt files for user admin (1 of 1): /admin/files/New text file.txt                                      
 [--->------------------------]                                                                             PHP Fatal error:  Call to a member function get() on null in /var/www/core-stable8.2/apps/files_sharing/lib/updater.php on line 76
PHP Stack trace:
PHP   1. {main}() /var/www/core-stable8.2/occ:0
PHP   2. require_once() /var/www/core-stable8.2/occ:11
PHP   3. OC\Console\Application->run() /var/www/core-stable8.2/console.php:79
PHP   4. Symfony\Component\Console\Application->run() /var/www/core-stable8.2/lib/private/console/application.php:100
PHP   5. Symfony\Component\Console\Application->doRun() /var/www/core-stable8.2/3rdparty/symfony/console/Symfony/Component/Console/Application.php:126
PHP   6. Symfony\Component\Console\Application->doRunCommand() /var/www/core-stable8.2/3rdparty/symfony/console/Symfony/Component/Console/Application.php:195
PHP   7. Symfony\Component\Console\Command\Command->run() /var/www/core-stable8.2/3rdparty/symfony/console/Symfony/Component/Console/Application.php:874
PHP   8. OC\Core\Command\Encryption\EncryptAll->execute() /var/www/core-stable8.2/3rdparty/symfony/console/Symfony/Component/Console/Command/Command.php:253
PHP   9. OCA\Encryption\Crypto\Encryption->encryptAll() /var/www/core-stable8.2/core/command/encryption/encryptall.php:122
PHP  10. OCA\Encryption\Crypto\EncryptAll->encryptAll() /var/www/core-stable8.2/apps/encryption/lib/crypto/encryption.php:446
PHP  11. OCA\Encryption\Crypto\EncryptAll->encryptAllUsersFiles() /var/www/core-stable8.2/apps/encryption/lib/crypto/encryptall.php:145
PHP  12. OCA\Encryption\Crypto\EncryptAll->encryptUsersFiles() /var/www/core-stable8.2/apps/encryption/lib/crypto/encryptall.php:202
PHP  13. OCA\Encryption\Crypto\EncryptAll->encryptFile() /var/www/core-stable8.2/apps/encryption/lib/crypto/encryptall.php:233
PHP  14. OC\Files\View->rename() /var/www/core-stable8.2/apps/encryption/lib/crypto/encryptall.php:255
PHP  15. OC_Hook::emit() /var/www/core-stable8.2/lib/private/files/view.php:729
PHP  16. call_user_func:{/var/www/core-stable8.2/lib/private/hook.php:103}() /var/www/core-stable8.2/lib/private/hook.php:103
PHP  17. OC\Files\Cache\Shared_Updater::renameHook() /var/www/core-stable8.2/lib/private/hook.php:103
PHP  18. OC\Files\Cache\Shared_Updater::moveShareToShare() /var/www/core-stable8.2/apps/files_sharing/lib/updater.php:61
```

**Comment by butonic** [comment]
fresh instance, master key enabled, uploaded a few files, nothing shared yet

**Comment by PVince81** [comment]
@schiesbn please have a look, potentially broken feature

**Comment by schiesbn** [comment]
Following @butonic description I can reproduce it... will have a look.

**Comment by PVince81** [comment]
@schiesbn thanks, let's hope it's an easy one

**Comment by schiesbn** [comment]
This is already fixed here: https://github.com/owncloud/core/pull/22159 and was released with 9.0, backports will be released with 8.2.4 and 8.1.7
